### PR TITLE
Upgrade prometheus crds for v0.53 compatibility

### PR DESCRIPTION
**Pull request by freecaykes**
## Description

Update Prometheus CRDs to be compatible with v0.53 of Prometheus Operator and Config Loader 

## For PR author

- [x] Tests for change.
- [x] If changing pkg/apis/, run `make gen-files`
- [x] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [x] Milestone set according to targeted release.
- [x] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: .semaphore/semaphore.yml
Comment (doublek) [inline .semaphore/semaphore.yml:5]: I think semaphore.yml should be reverted?

Comment (doublek) [inline .semaphore/semaphore.yml:5]: I think changes to semaphore.yml should be reverted?

File: Makefile
Comment (doublek) [inline Makefile:307]: nit `kubectl create`?

File: pkg/components/images.go
Comment (doublek) [inline pkg/components/images.go:28]: @hjiawei I don't think PrometheusRegistry is needed. Can you confirm please.

**Review by doublek — COMMENTED** [review]
Semaphore pipeline changes should be reverted, otherwise changes look good to me. Can you also update the PR description as to why we are introducing the two new `coreos-` "components".

File: config/enterprise_versions.yml
Comment (doublek) [inline config/enterprise_versions.yml:91]: nit: Missed this one. Looks like a partial comment?

File: pkg/components/enterprise.go
Comment (doublek) [inline pkg/components/enterprise.go:165]: nit: wonder if this should be just `ComponentCoreOSAlertmanager`

**Review by doublek — COMMENTED** [review]
Couple of more very minor nits. Sorry I missed them earlier.

**Review by doublek — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
